File: src/main/java/pl/edu/pg/AUI/product/service/api/CategoryService.java
package pl.edu.pg.AUI.product.service.api;

import pl.edu.pg.AUI.product.entity.Category;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface CategoryService {
    /**
     * Find category of given id
     * @param id category id
     * @return Optional container with category
     */
    Optional<Category> find(UUID id);

    /**
     * Create a new category
     * @param category new category
     */
    void create(Category category);

    /**
     * @return List of all available categories
     */
    List<Category> findAll();

    /**
     * Delete category of given id
     * @param id category id
     */
    void delete(UUID id);
}

File: aui-ng/src/app/category/service/category.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {Categories} from '../model/categories';
import {ProductDetails} from '../../product/model/product-details';
import {CategoryDetails} from '../model/category-details';
import {ProductForm} from '../../product/model/product-form';
import {CategoryForm} from '../model/category-form';

@Injectable({
  providedIn: 'root'
})
export class CategoryService {
  /**
   * @param http HTTP client
   */
  constructor(private http: HttpClient) { }

  /**
   * All categories
   */
  getCategories(): Observable<Categories> {
    return this.http.get<Categories>('api/categories')
  }

  getCategory(uuid: string): Observable<CategoryDetails> {
    return this.http.get<CategoryDetails>('/api/categories/' + uuid);
  }

  deleteCategory(uuid: string): Observable<any> {
    return this.http.delete('/api/categories/' + uuid);
  }

  putCategory(uuid: string, request: CategoryForm): Observable<any> {
    return this.http.put('/api/categories/' + uuid, request);
  }
}
